app/components/provider_user_notification_preferences_component.rb

```diff
@@ -0,0 +1,28 @@
+class ProviderUserNotificationPreferencesComponent < ViewComponent::Base
+  include ViewHelper
+  attr_reader :notification_preferences
+
+  CUSTOM_LEGENDS = {
+    application_withdrawn: 'Application withdrawn by candidate',
+    application_rejected_by_default: 'Application automatically rejected',
+  }.freeze
+
+  def initialize(notification_preferences)
+    @notification_preferences = notification_preferences
+  end
+
+  def on_off_options
+    [
+      OpenStruct.new(id: 'true', name: 'On'),
+      OpenStruct.new(id: 'false', name: 'Off'),
+    ]
+  end
+
+  def form_path
```

Comment:
can this be an attribute we pass into this component to avoid setting the path depending on where we call it?